SliceTracker/SliceTracker.py

```diff
@@ -3576,3 +3643,196 @@ def onCaseNumberChanged(self, caseNumber):
+    plot2 = self.chart.AddPlot(vtk.vtkChart.LINE)
+    plot2.SetInputData(self.chartTable, 0, 2)
+    plot2.SetColor(0, 255, 0, 255)
+    vtk.vtkPlotLine.SafeDownCast(plot2).SetMarkerStyle(4)
+    vtk.vtkPlotLine.SafeDownCast(plot2).SetMarkerSize(3 * plot2.GetPen().GetWidth())
+
+    plot3 = self.chart.AddPlot(vtk.vtkChart.LINE)
+    plot3.SetInputData(self.chartTable, 0, 3)
+    plot3.SetColor(0, 0, 255, 255)
+    vtk.vtkPlotLine.SafeDownCast(plot3).SetMarkerStyle(4)
+    vtk.vtkPlotLine.SafeDownCast(plot3).SetMarkerSize(3 * plot3.GetPen().GetWidth())
+
+    plot4 = self.chart.AddPlot(vtk.vtkChart.LINE)
+    plot4.SetInputData(self.chartTable, 0, 4)
+    plot4.SetColor(0, 0, 0, 255)
+    vtk.vtkPlotLine.SafeDownCast(plot4).SetMarkerStyle(4)
+    vtk.vtkPlotLine.SafeDownCast(plot4).SetMarkerSize(3 * plot4.GetPen().GetWidth())
+
+  def resetAndInitializeChart(self):
+    self.arrX.Initialize()
+    self.arrXD.Initialize()
+    self.arrYD.Initialize()
+    self.arrZD.Initialize()
+    self.arrD.Initialize()
+    self.addPlotPoints([[0, 0, 0], [0, 0, 0]])
+    if self.showLegendCheckBox.isChecked():
+      self.showLegendCheckBox.setChecked(False)
+
+  def onShowLegendChanged(self, checked):
+    if checked == 2:
+      self.chart.SetShowLegend(True)
+      self.chartView.scene().SetDirty(True)
+    else:
+      self.chart.SetShowLegend(False)
+      self.chartView.scene().SetDirty(True)
+
+  def onDockChartViewToggled(self, checked):
+    if checked:
+      self.chartPopupWindow = qt.QDialog()
+      self.chartPopupWindow.setWindowFlags(qt.Qt.WindowStaysOnTopHint)
+      layout = qt.QGridLayout()
+      self.chartPopupWindow.setLayout(layout)
+      layout.addWidget(self._chartView)
+      layout.addWidget(self.popupChartButton)
+      targetLayout = qt.QFormLayout()
+      targetLayout.addRow("Target:", self.DisplacementChartTargetSelector)
+      layout.addLayout(targetLayout, 3, 0)
+      self.chartPopupWindow.finished.connect(self.dockChartView)
+      self.chartPopupWindow.resize(self.chartPopupSize)
+      self.chartPopupWindow.move(self.chartPopupPosition)
+      self.chartPopupWindow.show()
+      self.popupChartButton.setText("Dock chart")
+      if self.DisplacementChartTargetSelector.currentText:
+        self._chartView.show()
+    else:
+      self.chartPopupWindow.close()
+
+  def dockChartView(self):
+    self.chartPopupSize = self.chartPopupWindow.size
+    self.chartPopupPosition = self.chartPopupWindow.pos
+    self.plottingFrameLayout.addWidget(self._chartView, 1, 0)
+    self.plottingFrameLayout.addWidget(self.popupChartButton, 2, 0)
+    child = self.targetSelectorLayout.takeAt(0)
+    self.targetSelectorLayout.insertRow(0, "Target:", self.DisplacementChartTargetSelector)
+    self.popupChartButton.setText("Undock chart")
+    self.popupChartButton.disconnect('toggled(bool)', self.onDockChartViewToggled)
+    self.popupChartButton.checked = False
+    self.popupChartButton.connect('toggled(bool)', self.onDockChartViewToggled)
```

Comment:
Instead of writing this:
```
self.popupChartButton.disconnect('toggled(bool)', self.onDockChartViewToggled)
self.popupChartButton.checked = False
self.popupChartButton.connect('toggled(bool)', self.onDockChartViewToggled)
```
I would write the following
```
self.popupChartButton.blockSignals(True)
self.popupChartButton.checked = False
self.popupChartButton.blockSignals(False)
```
This prevents connecting/disconnecting signals which in my opinion is not very clean. Maybe I did the same thing somewhere? ;)